### PR TITLE
Move scale management functions to a dedicated file

### PR DESCRIPTION
**Pull request by stockiNail**
This PR creates the scale module where there are some scale functions invoked from `annotation.js`.
These functions are NOT moved in a dedicated `helpers` or  in `helpers.chart` because they are invoked only from 1 point.

This is done to reduce the size of `annotation.js` and improving readability, concentrating all functions of the same topic in a dedicated module.

### CODE REVIEW

**Review by kurkle — DISMISSED** [review]
Those CC issues can and probably should be ignored here, because its hard to follow the history in GH when things move around and change in the same time. 

So IMO moving a function to new location should not change the function itself.

**Comment by stockiNail** [comment]
> Those CC issues can and probably should be ignored here, because its hard to follow the history in GH when things move around and change in the same time.
> 
> So IMO moving a function to new location should not change the function itself.

@kurkle ok, anyway I have already done. Let me push it, if you don't like it, I'll revert this update.

File: src/scale.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import {isFinite, valueOrDefault} from 'chart.js/helpers';
+
+export function adjustScaleRange(chart, scale, annotations) {
+  const range = getScaleLimits(scale, annotations);
+  let changed = changeScaleLimit(scale, range, 'min', 'suggestedMin') || changeScaleLimit(scale, range, 'max', 'suggestedMax');
```
Comment (kurkle) [inline src/scale.js:5]: The later function is not called when first one returns true. So this does not work properly.

I guess we don't have any tests that would catch this.

**Review by kurkle — COMMENTED** [review]
Contains a bug, but looks quite good.
I'd still like to have these changes in separate commit for the reasons stated previously.

File: src/scale.js (hunk 1 above)
Comment (stockiNail) [reply src/scale.js]: move to another PR

**Review by kurkle — APPROVED** [review]
